### PR TITLE
revisit examples

### PR DESCRIPTION
**Pull request by ryanSoley**
closes: #73

---

## What

**general updates/enhancements**

I came across a handful of small things that needed to be fixed to get all the examples running again:

  * updated the docs makefile to copy the examples to the appropriate subdirectories
    * this fixes a bug where the "view on GitHub" links aren't currently working
  * ran `pre-commit update` to get libraries to latest versions
  * removed `match_versions.sh` and moved the environment file to the repo's root
  * added CSS to make dataframes look good in the docs' dark mode
  * fixed a bug in the UI model where memory persistence with a default `root_dir` fails
    * I think it's another bug that the default `root_dir` ended up as `None` rather than `./`
  * fixed a bug where the async library needed `detail=True` added to a `fsspec` call

specific example notebook changes:

  * removed the auto-executing from all the examples in favor of just committing them executed
    * theres value in the outputs, so why not show them
  * quick look
    * made it a notebook and got rid of the RST version of it
      * I know I was the one who wanted them to be separate in the first place - that was dumb lol
    * removed some things that were more in-depth than a quick look
      * S3 sync, async, multiprocessing
  * collab and review
    * I just got rid of this one - I think we could really use a whole milestone around making the Intake stuff feel like a part of Rubicon rather than just a throwaway add on
  * logging basics
    * split the actual logging part into more than one cell
    * got rid of the dashboard stuff - not logging basics
  * logging concurrently
    * at some point this one stopped working because `multiprocessing` doesn't play nicely with iPython
    * I had to move the code into a script to get it to actually execute

## How to Test
  * build the new docs off this branch and check out the examples
  * run through all the example notebooks

### CODE REVIEW

File: notebooks/logging-examples/logging-basics.ipynb [hunk 1]
```diff
@@ -6,234 +6,577 @@
...
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "RandomForestClassifier accepts `n_estimators` (the number of trees in the forest) as a **parameter**. What if we wanted to tune this **parameter** to see how it affected our output **metric**, accuracy?\n",
-    "\n",
-    "This is where `Rubicon`'s **experiment** logging comes in handy. Let's get `Rubicon` setup:"
+    "With ``rubicon``, we can create a **project** to store as many **experiments** as we'd\n",
+    "like to run. Each of these **experiments** can hold important metadata about your\n",
```
Comment (joe-wolfe21) [inline notebooks/logging-examples/logging-basics.ipynb:276]: Nit: it switches from `we` and `our` to `you` and `your` in this example, might want to switch all one way

File: docs/source/conf.py [hunk 2]
```diff
@@ -48,6 +48,8 @@
 # See: https://nbsphinx.readthedocs.io/en/0.2.15/never-execute.html#Explicitly-Dis-/
 nbsphinx_execute = "never"
 nbsphinx_prolog = """
+|
```
Comment (joe-wolfe21) [inline docs/source/conf.py:51]: why is this needed?

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -24,11 +24,11 @@ Creating a Development Environment
 ==================================
 
 We have a ``conda`` environment YAML file with all the necessary dependencies
-in the ``ci`` directory.
+at the root of the repo.
```
Comment (joe-wolfe21) [inline docs/source/contributing.rst:27]: nit: repository.

File: notebooks/integrations/integration-git.ipynb [hunk 4]
```diff
@@ -69,18 +91,46 @@
...
+      "Dash app running on http://127.0.0.1:8050/\n"
+     ]
+    }
+   ],
    "source": [
     "from rubicon.ui import Dashboard\n",
     "\n",
-    "Dashboard(\"memory\").run_server()"
+    "\n",
+    "Dashboard(persistence=\"memory\").run_server()"
```
Comment (joe-wolfe21) [inline notebooks/integrations/integration-git.ipynb:123]: nit: the output is tough to read because of the background - we might want to override the css there

**Review by joe-wolfe21 — COMMENTED** [review]
Thanks for going through these and cleaning them up -- they look great! I left a few minor nits

File: docs/source/conf.py (hunk 2 above)
Comment (ryanSoley) [reply docs/source/conf.py:51]: it vertically centers the view on github link

File: notebooks/integrations/integration-git.ipynb (hunk 4 above)
Comment (ryanSoley) [reply notebooks/integrations/integration-git.ipynb:123]: I just hid all the error divs so that won't happen again

**Review by joe-wolfe21 — APPROVED** [review]
Nice cleanup!
